### ISSUE TITLE
pass current file as third argument?

### ISSUE THREAD

**Issue by felixroos** [shredsnews/rehype-urls#2]
Hello, and thanks for creating this plugin! It was exactly what I needed. I found it through [this comment](https://github.com/rehypejs/rehype-autolink-headings/issues/9#issuecomment-789545836)
There was a little workaround required for my use case, and I thought it could be generally helpful. 

I found myself needing to rewrite relative anchors to absolute ones, which only works when knowing the current path.
This is how I do it:

```js
const base = '/';
function absoluteAnchors() {
  return (tree, file) => {
    const chunks = file.history[0].split('/src/pages/'); // file.history[0] is the file path
    const path = chunks[chunks.length - 1].slice(0, -4); // only path inside src/pages, without .mdx
    return rehypeUrls((url) => {
      if (!url.href.startsWith('#')) {
        return;
      }
      const absoluteUrl = base + path + url.href;
      // console.log(url.href + ' -> ', absoluteUrl);
      return absoluteUrl;
    })(tree);
  };
}
``` 

(I needed this because I am using a base href, which results in all relative anchor links pointing to the root path instead of the current page)

**TLDR** It would be handy if the function passed to the plugin would also pass a third parameter the current file, like

```js
rehypeUrls((url, node, file) => /* ..*/);
```

I could also send a PR if you like

**Comment by brechtcs** [comment]
Sorry, this slipped through the cracks because I don't check Github as often as I used to.

It's been a while since I worked on any `unified` plugins, so the API is a bit hazy for me right now. If you already know how to implement it yourself though I'd be more than happy to accept a PR.
